lane.py

```diff
@@ -1923,18 +1940,19 @@ def works_from_database(self, _db, facets=None, pagination=None, **kwargs):
         lanes can be generated through search engine queries.
 
         :param _db: A database connection.
-        :param facets: A DatabaseBackedFacets object which may put additional
-           constraints on WorkList membership.
+        :param facets: A Facets object which may place additional
+           constraints on WorkList membership. Other objects are ignored.
         :param pagination: A Pagination object indicating which part of
            the WorkList the caller is looking at.
         :param kwargs: Ignored -- only included for compatibility with works().
         :return: A Query.
         """
-        if facets is not None and not isinstance(facets, DatabaseBackedFacets):
-            raise ValueError(
-                "Incompatible faceting object for DatabaseBackedWorkList: %r" %
+        if facets is not None and not isinstance(facets, Facets):
+            logging.warning(
+                "Ignoring incompatible faceting object for DatabaseBackedWorkList: %r",
```

Comment:
As I recall, you changed this from an exception to logging so that `Facets` could be passed in here. Now that `Facets` is explicitly handled, can we change back to raising an exception when (let's say) a `BaseFacets` is used? I think that indicates a programming error and an exception is appropriate.

Reply:
@leonardr,

We can end up with non-`Facets`-based subclasses of `BaseFacets` here (e.g., `FeatureFacets` and `MARCExporterFacets`). The latter comes through a code path that I know passes `Facets` objects, so not propagating `facets` is not a good option there.

For the former, we could just not propagate `facets` from `WorkList._featured_works_with_lanes`, but the word "presumably" in [this line](https://github.com/NYPL-Simplified/server_core/blob/master/lane.py#L1864) in the docstring gives me pause:
```
:param facets: A faceting object, presumably a FeaturedFacets
```
Anecdotally, I have not seen anything other than `FeatureFacets` coming through this method, but I'm not 100% sure.

These things together are why I thought it would be better to do the filtering once in `DatabaseBackedWorkList.works_from_database` and log the issue. I can also log a traceback here, which would make it possible to see who's doing what at some point in the future.

Reply:
We can also add a `BaseFacets.modify_database_query` implementation that's a no-op -- just returns `qu`. Then we wouldn't have to police what kind of faceting object you pass in. If it's a faceting object that doesn't know how to modify a database query, the query just doesn't get modified.

Reply:
The no-op `BaseFacets.modify_database_query` is the way to go, I think.